fcomm/tests/proof_tests.rs

```diff
@@ -66,12 +67,12 @@ fn test_prove_expression<T: AsRef<OsStr>>(
     cmd.assert().success();
 }
 
-fn test_open_commitment<T: AsRef<OsStr>>(
+fn test_open_commitment(
     mut cmd: Command,
     commitment: String,
-    input_path: T,
-    proof_path: T,
-    data_path: T,
+    input_path: &Utf8PathBuf,
+    proof_path: &Utf8PathBuf,
+    data_path: &Utf8PathBuf,
```

Comment:
So one of the ideas of #328 I believe was to use the `AsRef<&Path> for &Utf8Path` to focus the API on taking some `T: AsRef<&Path>`. 
Another would be to have those take an `&Utf8Path`, considering that `Utf8Path` is to rust's `str` what `Utf8PathBuf` is to `str`. 
This is test code, so it doesn't have to be the Alhambra, feel free to pick whichever you find simplest.

But the core idea is any time you're writing `&Utf8PathBuf`,  you can ask yourself the same questions as when you use `&String`. This may help:
https://blog.logrocket.com/understanding-rust-string-str/

TL;DR: this is using a path rather than defining a struct, so something that boils down to a reference into a string with path properties would suffice. i.e. `AsRef<&Path>` if we're formal, or `&Utf8Path`.